### PR TITLE
Use gradle-dep-tree with Audit

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Based on:
https://github.com/jfrog/build-info-go/pull/139
https://github.com/jfrog/jfrog-client-go/pull/719

### CODE REVIEW

File: xray/commands/audit/generic/auditmanager.go [hunk 1]
```diff
@@ -208,7 +215,7 @@ func auditMultipleWorkingDirs(params *Params) (results []services.ScanResponse,
 	}
 
 	if errorList.Len() > 0 {
-		err = errorutils.CheckError(errors.New(errorList.String()))
+		err = errors.New(errorList.String())
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:218]: This new error should be wrapped with CheckError.

File: xray/commands/audit/generic/auditmanager.go [hunk 2]
```diff
@@ -236,6 +243,7 @@ func doAudit(params *Params) (results []services.ScanResponse, isMultipleRoot bo
 			continue
 		}
 		techResults, e := audit.Audit(dependencyTrees, params.xrayGraphScanParams, params.serverDetails, params.progress, tech)
+		audit.BuildImpactPaths(techResults, params.dependencyTrees)
 		if e != nil {
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:247]: Let's put this error check right after the line that returns it.

Also, there's a missing error check at the bottom of this function. See these lines - 
```
	if errorList.Len() > 0 {
		err = errors.New(errorList.String())
	}
```

File: xray/commands/audit/generic/auditmanager.go [hunk 3]
```diff
@@ -274,25 +282,29 @@ func getTechDependencyTree(params *Params, tech coreutils.Technology) (dependenc
 		e = errorutils.CheckError(fmt.Errorf("%s is currently not supported", string(tech)))
 	}
 
-	return dependencyTrees, e
+	params.dependencyTrees = dependencyTrees
+	return services.FlattenGraph(dependencyTrees), e
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:286]: Should we actually flatten the graph, even if e is not nil?
Let's also rename `e` to `err` throughout this function.

File: xray/commands/audit/generic/auditmanager.go [hunk 4]
```diff
@@ -274,25 +282,29 @@ func getTechDependencyTree(params *Params, tech coreutils.Technology) (dependenc
...
+		javaProps = createMvnProps(params.DepsRepo(), params.ServerDetails())
 	}
 	return java.BuildDependencyTree(&java.DependencyTreeParams{
 		Tool:             tech,
 		InsecureTls:      params.insecureTls,
 		IgnoreConfigFile: params.ignoreConfigFile,
 		ExcludeTestDeps:  params.excludeTestDeps,
 		UseWrapper:       params.useWrapper,
-		JavaProps:        javaProps,
+		MvnProps:         javaProps,
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:300]: MvnProps --> JavaProps

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'm done going through everything.
I'd be happy to review this PR again following your changes.

File: xray/audit/java/gradle.go [hunk 5]
```diff
@@ -1,65 +1,268 @@
...
+type depTreeManager struct {
+	DependenciesTree
+	server       *config.ServerDetails
+	releasesRepo string
+	depsRepo     string
+	useWrapper   bool
+}
+
+// DependenciesTree represents a map between dependencies to their children dependencies in multiple projects.
+type DependenciesTree struct {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/java/gradle.go:58]: done

File: xray/audit/java/gradle.go [hunk 6]
```diff
@@ -1,65 +1,268 @@
...
+	tree map[string][]DependenciesPaths
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/java/gradle.go:59]: done

File: xray/audit/java/gradle.go [hunk 7]
```diff
@@ -1,65 +1,268 @@
...
 
-func buildGradleDependencyTree(excludeTestDeps, useWrapper, ignoreConfigFile bool, gradleConfigParams map[string]any) (dependencyTree []*services.GraphNode, err error) {
-	buildConfiguration, cleanBuild := createBuildConfiguration("audit-gradle")
-	defer cleanBuild(err)
+// The gradle-dep-tree generates a JSON representation for the dependencies for each gradle build file in the project.
+// parseDepTreeFiles iterates over those JSONs, and append them to the map of dependencies in DependenciesTree struct.
+func (dtp *depTreeManager) parseDepTreeFiles(jsonFiles []byte) error {
+	outputFiles := strings.Split(strings.TrimSpace(string(jsonFiles)), "\n")
+	for _, path := range outputFiles {
+		log.Info("reading", strings.TrimSpace(path))
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/java/gradle.go:72]: done

File: xray/audit/java/gradle.go [hunk 8]
```diff
@@ -1,65 +1,268 @@
...
+		tree, err := os.ReadFile(strings.TrimSpace(path))
+		if err != nil {
+			return err
```
[the remark this answers is not in the document]
Comment (omerzi) [reply xray/audit/java/gradle.go:75]: done

File: xray/commands/audit/generic/auditmanager.go [hunk 9]
```diff
@@ -240,46 +247,49 @@ func doAudit(params *Params) (results []services.ScanResponse, isMultipleRoot bo
...
 			Server:              params.serverDetails,
 			Tool:                pythonutils.PythonTool(tech),
 			RemotePypiRepo:      params.depsRepo,
 			PipRequirementsFile: params.requirementsFile})
 	case coreutils.Nuget:
-		dependencyTrees, e = nuget.BuildDependencyTree()
+		dependencyTrees, err = nuget.BuildDependencyTree()
 	default:
-		e = errorutils.CheckError(fmt.Errorf("%s is currently not supported", string(tech)))
+		err = errorutils.CheckError(fmt.Errorf("%s is currently not supported", string(tech)))
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:282]: `errorutils.CheckError(fmt.Errorf` can be simplified to `errorutils.CheckErrorf`

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**